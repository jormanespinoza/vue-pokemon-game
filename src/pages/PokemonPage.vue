<template>
  <span v-if="!pokemon">Looking for a pokemon, please wait...</span>

  <div v-else>
    <h1>Who is this pokemon?</h1>

    <PokemonPicture 
      :pokemon-id="pokemon.id"
      :show-pokemon="showPokemon"
    />

    <PokemonOptions
      :pokemons="pokemons"
      @pokemon-selected="checkAnswer($event)"
    />

    <template v-if="showAnswer">
      <span 
        class="fade-in answer" 
        :class="correctAnswer ? 'correct' : 'wrong'"
      >
        {{ answer }}
      </span>
      <button @click="newGame">New game</button>
    </template>
  </div>
</template>

<script>
import PokemonPicture from '@/components/PokemonPicture'
import PokemonOptions from '@/components/PokemonOptions'

import getPokemonOptions from '@/helpers/getPokemonOptions'

export default {
  components: {
    PokemonPicture,
    PokemonOptions
  },
  data() {
    return {
      pokemons: [],
      pokemon: null,
      showPokemon: false,
      showAnswer: false,
      correctAnswer: false,
      answer: ''
    }
  },
  methods: {
    async mixPokemons() {
      this.pokemons = await getPokemonOptions()
      this.pokemon = this.pokemons[Math.floor(Math.random() * 4)];
    },
    checkAnswer(pokemonSelectedId) {
      this.showPokemon = true
      this.answer = (this.pokemon.id === pokemonSelectedId)
        ? `Correct, ${this.pokemon.name}!`
        : `Oops, it was ${this.pokemon.name}!`
      this.correctAnswer = this.pokemon.id === pokemonSelectedId
      this.showAnswer = true
    },
    newGame() {
      this.showPokemon = false
      this.showAnswer = false
      this.pokemon = null
      this.mixPokemons()
    }
  },
  mounted() {
    this.mixPokemons()
  }
}
</script>

<style scope>
.answer {
  display: block;
  font-weight: bold;
}

.answer.correct {
  color: #2d649b;
}

.answer.wrong {
  color: #da4f4f;
}

button {
  margin-top: 15px;
}
</style>