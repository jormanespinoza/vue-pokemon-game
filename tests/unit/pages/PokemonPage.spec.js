import { shallowMount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'

describe('PokemonPage.vue', () => {

    test('should match snapshot', () => {
        const wrapper = shallowMount(PokemonPage)

        expect(wrapper.html()).toMatchSnapshot()
    })
})