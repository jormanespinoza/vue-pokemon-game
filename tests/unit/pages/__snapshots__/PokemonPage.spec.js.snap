// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage.vue should match snapshot 1`] = `<span>Looking for a pokemon, please wait...</span>`;
